internal/pkg/dependencies/template.go

```diff
@@ -77,16 +77,10 @@ func (c *common) Template(reference model.TemplateRef) (*template.Template, erro
 		return nil, err
 	}
 
-	// Get template
-	templateRecord, found := repository.GetById(reference.TemplateId())
-	if !found {
-		return nil, fmt.Errorf(`template "%s" not found`, reference.TemplateId())
-	}
-
-	// Get version
-	versionRecord, found := templateRecord.GetByVersion(reference.Version())
-	if !found {
-		return nil, fmt.Errorf(`template "%s" found, but version "%s" is missing`, reference.TemplateId(), reference.Version().Original())
+	// Get template version
+	versionRecord, err := repository.GetTemplateVersion(reference.TemplateId(), reference.Version())
```

Comment:
Vytvorena nova metoda `GetTemplateVersion`, aby nebol duplicitny kod ... kedze okrem tohto miesta to bude aj v tvojom kode.